File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        L,R=0,0
        n=len(s)
        found=set()
        maxV=0
        while L<n and R<n:
            if s[R] in found:
                while s[R] in found:
                    found.remove(s[L])
                    L+=1
            else:
                maxV=max(R-L+1,maxV)
            found.add(s[R])
            R+=1
        return maxV

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<char> found;

        int L = 0, R = 0, n = s.size();
        int maxV = 0;
        while (L < n && R < n) {
            while (found.find(s[R]) != found.end()) 
                found.erase(s[L++]);
            maxV = max(R - L + 1, maxV);
            found.insert(s[R]);
            R += 1;
        }
        return maxV;
    }
};